server/server.go

```diff
@@ -222,9 +229,9 @@ func setupSubsonic(r *mux.Router, ctrl *ctrlsubsonic.Controller) {
 	r.Handle("/getSimilarSongs2{_:(?:\\.view)?}", ctrl.H(ctrl.ServeGetSimilarSongsTwo))
 
 	// raw
-	r.Handle("/download{_:(?:\\.view)?}", ctrl.HR(ctrl.ServeDownload))
 	r.Handle("/getCoverArt{_:(?:\\.view)?}", ctrl.HR(ctrl.ServeGetCoverArt))
 	r.Handle("/stream{_:(?:\\.view)?}", ctrl.HR(ctrl.ServeStream))
+	r.Handle("/download{_:(?:\\.view)?}", ctrl.HR(ctrl.ServeStream))
```

Comment:
yeah very nice point. i really wasn't sure about this. but what i find annoying is that when i'm on a train streaming music, dsub will transcode the first track that i select so it stream nice and quick. but then for the tracks that it caches ahead, it uses download.view. so since it's downloading straight flacs, it just hangs

you're right that download should just serve the raw file so i'll put it back. but any ideas how to fix my issue? :grin:

Reply:
Hmm, maybe you should file a bug report to dSub developer(s)? :)
In my case, iSub (yet another SubSonic API client for iOS, this time with Opus support) preloads next track (and pre-caches tracks when I tap a "download to cache" button on a playlist) using `/stream`, as expected. IIRC, some Android client I used to use had different settings for Wi-Fi and 3G/LTE/etc modes, and the default setting was to download and (pre-)cache unmodified tracks in all cases.